File: libraries/FusionChartsFree/Code/VBNET/App_Code/DbConn.vb
Imports Microsoft.VisualBasic
Imports System.Data.Odbc
Imports System.Data
Imports System.Web
Imports System.Configuration

Namespace DataConnection

    ''' <summary>
    ''' DataBase Connection Class.
    ''' </summary>
    Public Class DbConn
        Public connection As OdbcConnection
        Public ReadData As OdbcDataReader
        Public aCommand As OdbcCommand

        ''' <summary>
        ''' Data Connection and get Data Reader
        ''' </summary>
        ''' <param name="strQuery">SQL Query</param>
        Public Sub New(ByVal strQuery As String)
            Dim ConnectionString As String, connectionName As String

            ' MS Access DataBase Connection - Defined in Web.Config
            connectionName = "MSAccessConnection"

            ' SQL Server DataBase Connection - Defined in Web.Config
            ' connectionName = "SQLServerConnection";

            ' Creating Connection string using web.config connection string
            ConnectionString = ConfigurationManager.ConnectionStrings(connectionName).ConnectionString
            Try

                ' Creating OdbcConnection Oject
                connection = New OdbcConnection()

                ' Setting Conection String
                connection.ConnectionString = ConnectionString

                ' Open Connection
                connection.Open()

                ' get reader
                GetReader(strQuery)

            Catch ex As Exception
                HttpContext.Current.Response.Write(ex.Message)
            End Try

        End Sub

        ''' <summary>
        ''' Create an instance dataReader
        ''' </summary>
        ''' <param name="strQuery">SQL Query</param>
        ''' <remarks>Return type object of OdbcDataReader</remarks>
        Public Sub GetReader(ByVal strQuery As String)

            '  Create a Command object
            aCommand = New OdbcCommand(strQuery, connection)

            ' Create data reader object using strQuery string
            ReadData = aCommand.ExecuteReader(CommandBehavior.CloseConnection)

        End Sub
    End Class
End Namespace

File: libraries/FusionChartsFree/Code/CSNET/App_Code/DbConn.cs
using System;
using System.Data;
using System.Data.Odbc;
using System.Web;
using System.Configuration;

namespace DataConnection
{
    /// <summary>
    /// DataBase Connection Class.
    /// </summary>
    public class DbConn
    {

        //  Create a database Connection. using here Access Database
        //  Return type object of OdbcConnection

        public OdbcConnection connection;
        public OdbcDataReader ReadData;
        public OdbcCommand aCommand;
        /// <summary>
        /// Data Connection and get Data Reader
        /// </summary>
        /// <param name="strQuery">SQL Query</param>
        public DbConn(string strQuery)
        {
            // MS Access DataBase Connection - Defined in Web.Config
            string connectionName = "MSAccessConnection";

            // SQL Server DataBase Connection - Defined in Web.Config
            //string connectionName = "SQLServerConnection";

            // Creating Connection string using web.config connection string
            string ConnectionString = ConfigurationManager.ConnectionStrings[connectionName].ConnectionString;
            try
            {
                // create connection object
                connection = new OdbcConnection();
                // set connection string
                connection.ConnectionString = ConnectionString;
                // open connection
                connection.Open();
                // get reader
                GetReader(strQuery);
            }
            catch (Exception e)
            {
                HttpContext.Current.Response.Write(e.Message.ToString());
            }

        }

        // Create an instance dataReader
        // Return type object of OdbcDataReader
        /// <summary>
        /// Get Data Reader
        /// </summary>
        /// <param name="strQuery">SQL Query</param>
        public void GetReader(string strQuery)
        {
            //  Create a Command object
            aCommand = new OdbcCommand(strQuery, connection);

            // Create data reader object using strQuery string
            // Auto close connection
            ReadData = aCommand.ExecuteReader(CommandBehavior.CloseConnection);

        }

    }
}